pallets/runtime/tests/src/utility_test.rs

```diff
@@ -228,3 +231,91 @@ fn _relay_unhappy_cases() {
+    let alice_did = register_keyring_account_with_balance(AccountKeyring::Alice, 1_000).unwrap();
+    let bob_key = AccountKeyring::Bob.public();
+    let bob_origin = Origin::signed(bob_key);
+    let bob_signer = Signatory::Account(bob_key);
+
+    add_secondary_key(alice_did, bob_signer);
+    let low_risk_name: PortfolioName = b"low risk".into();
+    assert_ok!(Portfolio::create_portfolio(
+        bob_origin.clone(),
+        low_risk_name.clone()
+    ));
+    assert_last_event!(EventTest::portfolio(pallet_portfolio::RawEvent::PortfolioCreated(_, _, _)));
+    assert_eq!(
+        Portfolio::portfolios(&alice_did, &PortfolioNumber(1)),
+        low_risk_name.clone()
+    );
+    let bob_pallet_permissions = vec![
+        PalletPermissions::new(b"Identity".into(), SubsetRestriction(None)),
+        PalletPermissions::new(
+            b"Portfolio".into(),
+            SubsetRestriction::elems(vec![
+                b"move_portfolio_funds".into(),
+                b"rename_portfolio".into(),
+            ]),
+        ),
+    ];
+    let bob_permissions = Permissions {
+        asset: SubsetRestriction::default(),
+        extrinsic: SubsetRestriction(Some(bob_pallet_permissions.into_iter().collect())),
+        portfolio: SubsetRestriction::default(),
+    };
+    assert_ok!(Identity::set_permission_to_signer(
+        alice_origin,
+        bob_signer,
+        bob_permissions,
+    ));
+    let bob_secondary_key = &Identity::did_records(&alice_did).secondary_keys[0];
+    assert_eq!(
+        true,
+        bob_secondary_key
+            .has_extrinsic_permission(&b"Portfolio".into(), &b"rename_portfolio".into())
+    );
+    assert_eq!(
+        false,
+        bob_secondary_key
+            .has_extrinsic_permission(&b"Portfolio".into(), &b"create_portfolio".into())
+    );
+    let high_risk_name: PortfolioName = b"high risk".into();
+    assert_err!(
+        Portfolio::create_portfolio(bob_origin.clone(), high_risk_name.clone()),
+        pallet_permissions::Error::<TestStorage>::UnauthorizedCaller
+    );
+    let calls = vec![
+        Call::Portfolio(PortfolioCall::create_portfolio(high_risk_name.clone())),
+        Call::Portfolio(PortfolioCall::rename_portfolio(
+            1.into(),
+            high_risk_name.clone(),
+        )),
+    ];
+    assert_ok!(Utility::batch(bob_origin.clone(), calls.clone()));
+    assert_event_doesnt_exist!(EventTest::pallet_utility(Event::BatchCompleted));
+    assert_event_exists!(EventTest::pallet_utility(Event::BatchInterrupted(_, _)));
+    assert_eq!(
+        Portfolio::portfolios(&alice_did, &PortfolioNumber(1)),
+        low_risk_name
+    );
+    assert_ok!(Utility::batch_optimistic(bob_origin, calls));
+    assert_event_exists!(
+        EventTest::pallet_utility(Event::BatchOptimisticFailed(errors)),
+        errors.len() == 1
+    );
+    assert_eq!(
+        Portfolio::portfolios(&alice_did, &PortfolioNumber(1)),
+        high_risk_name
+    );
```

Comment:
```suggestion
    check_name(high_risk_name);
```
let's turn this logic into a closure to dedup?